### PR TITLE
chore: prevent dev server from reloading unnecessarily

### PR DESCRIPTION
**Pull request by AndrewADev**
I noticed recently while using the dev server, that it was reloading frequently due to changes to unrelated files. In this case, for updates to git local state performed in the background:
image
_screenshot of dev server output to the console_

These changes update the dev server settings to reduce unnecessary refreshes.

I tested a few common scenarios (edits to `.ts, .html, .json` files), and it worked as expected (also that edits to files like `.md` no longer trigger reloads). Let me know though, if you notice any cases where it is no longer updating when it should.

### CODE REVIEW

**Comment by JamesLMilner** [comment]
I'm just on mobile at the moment but will hopefully get to this later this week! This looks like it would be a good improvement for sure - I just need to double check if there's any files missing.

File: development/webpack.config.js [hunk 1]
```diff
@@ -24,9 +24,14 @@ module.exports = {
...
 		static: {
-			directory: path.join(__dirname),
+			directory: path.join(__dirname, "public"),
 		},
-		watchFiles: ["../**/**"],
+		watchFiles: [
+			"./src",
+			"./*.{js,json,ts,html}",
+			"../src",
+			"../*.{js,json,ts,html}",
```
Comment (JamesLMilner) [inline development/webpack.config.js:33]: Lets add `./.env` in here too so environment variable files are caught

**Comment by JamesLMilner** [comment]
One comment but otherwise looks good!

File: development/webpack.config.js (hunk 1 above)
Comment (AndrewADev) [reply development/webpack.config.js:33]: It doesn't seem that runtime changes to the values of the variables are supported by the `dotenv-webpack` plugin, currently. If I add `.env` to the watched files, the dev server will reload for changes to the `.env` file, but the values of any changed variables are not updated in the app until after restart.

See the current open issue here: https://github.com/mrsteele/dotenv-webpack/issues/476

Comment (JamesLMilner) [reply development/webpack.config.js:33]: Ah! Fair enough - I looked at the original thread and looks like there's a solution but we can do that separately another time. This looks like it is an improvement regardless so let's get it in 🙂

**Pull request merged**